scheduler/util.go

```diff
@@ -424,6 +430,48 @@ func connectServiceUpdated(servicesA, servicesB []*structs.Service) comparison {
 	return same
 }
 
+func volumeMountsUpdated(a, b []*structs.VolumeMount) comparison {
+	setA := set.HashSetFrom(a)
+	setB := set.HashSetFrom(b)
+
+	if setA.Equal(setB) {
+		return same
+	}
+
+	bVMMap := helper.SliceToMap[map[string]*structs.VolumeMount](b,
+		func(vm *structs.VolumeMount) string {
+			return vm.Volume
+		})
+
+	for _, atvm := range a {
+		if c := volumeMountUpdated(atvm, bVMMap[atvm.Volume]); c.modified {
+			return c
+		}
+		delete(bVMMap, atvm.Volume)
+	}
+
+	if len(bVMMap) > 0 {
+		return difference("volume mount added", nil, bVMMap)
+	}
```

Comment:
What I was trying to get at before is that all of this can now be eliminated in favor of just

```
return difference("volume mounts changed")
```

the `comparison` we create is nothing more than a debugging tool; if you get the "volume mounts changed" differential the person debugging should be able to figure it out by just looking at the submitted job(s).

Reply:
not sure I follow what you mean

Reply:
The only thing we care about from `volumeMountsUpdated` function is whether or not volume mounts were modified (while ignoring sort order). All of the logic for determining _which_ volume mount changed is extraneous / bug prone.

Since we don't care about _which_ volume mount changed, we can delete all this code and just use the `Equal` method between the two sets, in which case this function simplifies to something like,

```
func volumeMountsUpdated(a, b []*structs.VolumeMount) comparison {
	setA := set.HashSetFrom(a)
	setB := set.HashSetFrom(b)

	if setA.Equal(setB) {
		return same
	}

    return difference("volume mounts changed")
}